core/src/main/java/com/oath/oak/Chunk.java

```diff
@@ -1156,10 +1175,28 @@ private void findNewNextInStack() {
             }
         }
 
+        private void pushToStack(boolean compareWithPrevAnchor) {
+            while (next != Chunk.NONE) {
```

Comment:
check if changing this to next != prevAnchor will save the keys bytebuffer comparisons